pkg/virt-controller/watch/vm.go

```diff
@@ -3157,13 +3155,15 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 		}
 
 		if syncErr == nil {
-			if !equality.Semantic.DeepEqual(vm, vmCopy) {
+			if !equality.Semantic.DeepEqual(vm.Spec, vmCopy.Spec) || !equality.Semantic.DeepEqual(vm.ObjectMeta, vmCopy.ObjectMeta) {
 				updatedVm, err := c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy, metav1.UpdateOptions{})
 				if err != nil {
 					syncErr = &syncErrorImpl{fmt.Errorf("Error encountered when trying to update vm according to add volume and/or memory dump requests: %v", err), FailedUpdateErrorReason}
 				} else {
 					vm = updatedVm
 				}
+			} else {
+				vm = vmCopy
```

Comment:
That means that if spec/objectmeta changed, we ignore status changes, right?
Example: something above changed both vm.spec and vm.status. We'll call update, which will lose the status changes, and return updatedVm instead of vmCopy.
How about something like that instead?
```suggestion
			}
				vm.Status = *vmCopy.Status
```

Reply:
>That means that if spec/objectmeta changed, we ignore status changes, right?

Yes, there are couple of edge cases that are not fixed by this PR but the last 3 PRs are making it much more simple to fix these cases. @acardace should have a PR that will fix at least partially these cases.

>Example: something above changed both vm.spec and vm.status. We'll call update, which will lose the status changes, and return updatedVm instead of vmCopy.
How about something like that instead?

WDYT about adding `equality.Semantic.DeepEqual(vm.Status, vmCopy.Status) && (!equality.Semantic.DeepEqual(vm.Spec, vmCopy.Spec) || !equality.Semantic.DeepEqual(vm.ObjectMeta, vmCopy.ObjectMeta)) {` ? But as I said this should be fixed as follow up.


The `vm.Status = *vmCopy.Status` should not be needed because if the status have changed then no spec changes should be done as well...

Reply:
> The `vm.Status = *vmCopy.Status` should not be needed because if the status have changed then no spec changes should be done as well...

If we know we never change both Status and Spec at the same time, then this is all good!
/approve

Reply:
We do unfortunately but that is bug on its own but the PR doesn't make it worse